doc/spec.md

```diff
@@ -1418,7 +1437,7 @@ Type parameter names must be unique. A compile-time error occurs if two or more
 
 The scope of a type parameter extends over the entire declaration with which the type parameter list is associated, with the exception of static member declarations in classes.
 
-Each type parameter has an associated type parameter ***constraint*** that establishes an upper bound for type arguments. Omitting a constraint corresponds to specifying the empty object type `{}`. Type parameters declared in a particular type parameter list may not be referenced in constraints in that type parameter list.
+Each type parameter has an associated type parameter ***constraint*** that establishes an upper bound for type arguments. Omitting a constraint or specifying type `any` as the constraint corresponds to specifying the empty object type `{}`. Type parameters declared in a particular type parameter list may not be referenced in constraints in that type parameter list.
```

Comment:
It's always been allowed (any type can be a constraint because a constraint can be a type parameter). We now use it to indicate that additional properties are expected. See the last example in [this section](https://github.com/Microsoft/TypeScript/blob/updateLangSpec/doc/spec.md#3.11.5).